### PR TITLE
Magic prompt blocklist

### PR DESCRIPTION
**Pull request by akx**
This enables magic prompt blocklisting with dynamicprompts 0.4.0.

### CODE REVIEW

File: scripts/dynamic_prompting.py [hunk 1]
```diff
@@ -196,6 +180,13 @@ def ui(self, is_img2img):
                             elem_id="magic-prompt-model",
                         )
 
+                        magic_blocklist_regex = gr.Textbox(
+                            label="Magic prompt blocklist regex",
+                            value="",
+                            elem_id="magic-prompt-blocklist-regex",
+                            placeholder="Regular expression for blocking terms out of the generated prompt",
```
Comment (adieyal) [inline scripts/dynamic_prompting.py:187]: Could you add an example regex so that less technical users can get a sense of how to use it?

**Comment by akx** [comment]
@adieyal Can you release a new version of `dynamicprompts` now that #8 was merged in? :)

**Comment by adieyal** [comment]
> @adieyal Can you release a new version of `dynamicprompts` now that [adieyal/dynamicprompts#8](https://github.com/adieyal/dynamicprompts/pull/8) was merged in? :)

Done, v0.4.0

**Comment by axemaster** [comment]
What is blocklisting? (some people use this as a synonym for blacklisting btw, I assume you're not).

**Comment by adieyal** [comment]
The rationale is that there are some keywords you don't want in your prompts. I think Greg Rutkowski was the main use case - since Magic Prompts was trained on real prompts, you'll find a lot of Greg Rutkowski in here. If you prefer your prompts Rutkowski free, then block list that term.

**Comment by axemaster** [comment]
Oh that makes sense. Thanks for explaining. Definitely too much Greg Rutkowski in the prompt universe. I read that he doesn't like it either.

**Comment by axemaster** [comment]
Found it.

`./extensions/stable-diffusion-webui-Prompt_Generator/blacklist.txt`

Do wildcards work in there? e.g. "style of *", "* from artgem"

This could be doc'd in the main README for the extension. :) I didn't see it there.

**Comment by axemaster** [comment]
Looking at the commit I guess they are regular expressions

**Comment by axemaster** [comment]
I see it's doc'd in the UI. Prompt magic doesn't work on my Mac M1 so I never noticed that. ¯\_(ツ)_/¯

**Comment by akx** [comment]
@axemaster [`stable-diffusion-webui-Prompt_Generator` is not this extension](https://github.com/imrayya/stable-diffusion-webui-Prompt_Generator), though.

But yes, `blocklist` here is a synonym for `blacklist`.

**Pull request merged**